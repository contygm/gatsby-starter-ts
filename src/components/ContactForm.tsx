import React from 'react';

/**
 * @description Page header for standard pages
 * @param title - the page title to be displayed
 * @param alignCenter - boolean;
 */
export const ContactForm = () => {
    // console.log form values on submit
    function onSubmit(e: React.SyntheticEvent) {
        e.preventDefault();

        // const target = e.target as typeof e.target & {
        //     email: { value: string };
        //     message: { value: string };
        //     name: { value: string };
        // };

        console.log('contact form submit');
        // console.log('email:', target.email.value);
        // console.log('message:', target.message.value);
    }

    // mailto action will only work in IE (lol)
    return (
        <form
            className="is-small"
            method="get"
            // action={`mailto:${props.email}`}
            encType="text/plain"
            onSubmit={onSubmit}
        >
            <div className="field">
                <label className="label">Name</label>
                <div className="control">
                    <input
                        className="input"
                        type="text"
                        placeholder="Name"
                        name="name"
                    />
                </div>
            </div>

            <div className="field">
                <label className="label">YourEmail</label>
                <div className="control">
                    <input
                        className="input"
                        type="text"
                        placeholder="email@email.com"
                        name="email"
                    />
                </div>
            </div>

            <div className="field">
                <label className="label">Message</label>
                <div className="control">
                    <textarea
                        className="textarea"
                        placeholder="Your messages"
                        name="message"
                    ></textarea>
                </div>
            </div>

            <div className="field is-grouped">
                <div className="control">
                    <button
                        type="submit"
                        className="button is-primary"
                    >
                        Submit
                    </button>
                </div>
                <div className="control">
                    <button
                        type="reset"
                        name="reset"
                        className="button has-background-grey-lighter"
                    >
                        Cancel
                    </button>
                </div>
            </div>
        </form>
    );
};
