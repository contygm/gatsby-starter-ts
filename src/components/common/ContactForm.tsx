import React from 'react';

/**
 * A basic contact form. Fields: name, email, message, submit, and cancel.
 *
 * Currently, there is no submission action.
 * @category Components
 */
export const ContactForm = () => {
    // console.log form values on submit
    // TODO: make an alert
    function onSubmit(e: React.SyntheticEvent) {
        e.preventDefault();

        // const target = e.target as typeof e.target & {
        //     email: { value: string };
        //     message: { value: string };
        //     name: { value: string };
        // };

        console.log('contact form submit');
        // console.log('email:', target.email.value);
        // console.log('message:', target.message.value);
    }

    // mailto action will only work in IE (lol)
    return (
        <form
            className="is-small"
            method="get"
            // action={`mailto:${props.email}`}
            encType="text/plain"
            onSubmit={onSubmit}
        >
            <div className="field">
                <label className="label">Name</label>
                <div className="control">
                    <input
                        className="input"
                        type="text"
                        placeholder="Name"
                        name="name"
                    />
                </div>
            </div>

            <div className="field">
                <label className="label">YourEmail</label>
                <div className="control">
                    <input
                        className="input"
                        type="text"
                        placeholder="email@email.com"
                        name="email"
                    />
                </div>
            </div>

            <div className="field">
                <label className="label">Message</label>
                <div className="control">
                    <textarea
                        className="textarea"
                        placeholder="Your messages"
                        name="message"
                    ></textarea>
                </div>
            </div>

            <div className="contact-action-buttons">
                <div className="control">
                    <button
                        type="submit"
                        className="primary-button"
                    >
                        Submit
                    </button>
                </div>
                <div className="control">
                    <button
                        type="reset"
                        name="reset"
                        className="light-grey-button"
                    >
                        Cancel
                    </button>
                </div>
            </div>
        </form>
    );
};
