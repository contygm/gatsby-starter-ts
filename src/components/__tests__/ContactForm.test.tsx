import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import ContactForm from '../ContactForm';

describe('ContactForm', () => {
    it('renders correctly', () => {
        const { asFragment } = render(<ContactForm />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('submit does nothing', () => {
        const { asFragment, getByText } = render(<ContactForm />);

        expect(asFragment()).toMatchSnapshot();
        fireEvent.click(getByText("Submit"));
        expect(asFragment()).toMatchSnapshot();
    });
});
