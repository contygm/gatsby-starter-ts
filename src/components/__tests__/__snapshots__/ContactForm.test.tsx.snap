// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm renders correctly 1`] = `
<DocumentFragment>
  <form
    class="is-small"
    enctype="text/plain"
    method="get"
  >
    <div
      class="field"
    >
      <label
        class="label"
      >
        Name
      </label>
      <div
        class="control"
      >
        <input
          class="input"
          name="name"
          placeholder="Name"
          type="text"
        />
      </div>
    </div>
    <div
      class="field"
    >
      <label
        class="label"
      >
        YourEmail
      </label>
      <div
        class="control"
      >
        <input
          class="input"
          name="email"
          placeholder="email@email.com"
          type="text"
        />
      </div>
    </div>
    <div
      class="field"
    >
      <label
        class="label"
      >
        Message
      </label>
      <div
        class="control"
      >
        <textarea
          class="textarea"
          name="message"
          placeholder="Your messages"
        />
      </div>
    </div>
    <div
      class="field is-grouped"
    >
      <div
        class="control"
      >
        <button
          class="button is-primary"
          type="submit"
        >
          Submit
        </button>
      </div>
      <div
        class="control"
      >
        <button
          class="button has-background-grey-lighter"
          name="reset"
          type="reset"
        >
          Cancel
        </button>
      </div>
    </div>
  </form>
</DocumentFragment>
`;

exports[`ContactForm submit does nothing 1`] = `
<DocumentFragment>
  <form
    class="is-small"
    enctype="text/plain"
    method="get"
  >
    <div
      class="field"
    >
      <label
        class="label"
      >
        Name
      </label>
      <div
        class="control"
      >
        <input
          class="input"
          name="name"
          placeholder="Name"
          type="text"
        />
      </div>
    </div>
    <div
      class="field"
    >
      <label
        class="label"
      >
        YourEmail
      </label>
      <div
        class="control"
      >
        <input
          class="input"
          name="email"
          placeholder="email@email.com"
          type="text"
        />
      </div>
    </div>
    <div
      class="field"
    >
      <label
        class="label"
      >
        Message
      </label>
      <div
        class="control"
      >
        <textarea
          class="textarea"
          name="message"
          placeholder="Your messages"
        />
      </div>
    </div>
    <div
      class="field is-grouped"
    >
      <div
        class="control"
      >
        <button
          class="button is-primary"
          type="submit"
        >
          Submit
        </button>
      </div>
      <div
        class="control"
      >
        <button
          class="button has-background-grey-lighter"
          name="reset"
          type="reset"
        >
          Cancel
        </button>
      </div>
    </div>
  </form>
</DocumentFragment>
`;

exports[`ContactForm submit does nothing 2`] = `
<DocumentFragment>
  <form
    class="is-small"
    enctype="text/plain"
    method="get"
  >
    <div
      class="field"
    >
      <label
        class="label"
      >
        Name
      </label>
      <div
        class="control"
      >
        <input
          class="input"
          name="name"
          placeholder="Name"
          type="text"
        />
      </div>
    </div>
    <div
      class="field"
    >
      <label
        class="label"
      >
        YourEmail
      </label>
      <div
        class="control"
      >
        <input
          class="input"
          name="email"
          placeholder="email@email.com"
          type="text"
        />
      </div>
    </div>
    <div
      class="field"
    >
      <label
        class="label"
      >
        Message
      </label>
      <div
        class="control"
      >
        <textarea
          class="textarea"
          name="message"
          placeholder="Your messages"
        />
      </div>
    </div>
    <div
      class="field is-grouped"
    >
      <div
        class="control"
      >
        <button
          class="button is-primary"
          type="submit"
        >
          Submit
        </button>
      </div>
      <div
        class="control"
      >
        <button
          class="button has-background-grey-lighter"
          name="reset"
          type="reset"
        >
          Cancel
        </button>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
